Agents/FeatureInfoAgent/code/src/main/java/com/cmclinnovations/featureinfo/FeatureInfoAgent.java

```diff
@@ -0,0 +1,417 @@
+package com.cmclinnovations.featureinfo;
+
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.core.Response;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.stereotype.Controller;
+
+import com.cmclinnovations.featureinfo.config.ConfigEndpoint;
+import com.cmclinnovations.featureinfo.config.ConfigStore;
+import com.cmclinnovations.featureinfo.config.EndpointType;
+import com.cmclinnovations.featureinfo.kg.ClassHandler;
+import com.cmclinnovations.featureinfo.kg.MetaHandler;
+import com.cmclinnovations.featureinfo.kg.TimeHandler;
+
+import uk.ac.cam.cares.jps.base.agent.JPSAgent;
+import uk.ac.cam.cares.jps.base.discovery.AgentCaller;
+import uk.ac.cam.cares.jps.base.query.RemoteStoreClient;
+import uk.ac.cam.cares.jps.base.timeseries.TimeSeriesClient;
+
+/**
+ * This agent expects a HTTP request containing a JSON string with the "iri" and "endpoint" for an individual
+ * feature. This information is then used (in conjunction with hardcoded SPARQL queries) to gather data on
+ * that feature and return it as a JSON object.
+ * 
+ * Note: this version of the agent is a VERY rough implementation and should not be used outside of very specific
+ * use cases. It currently relies on hardcoded SPARQL queries, and will not work out-of-the box for any old data sets.
+ * Future improvements will make this more generic, at which point the agent could be used more widely.
+ *
+ * @author Michael Hillman {@literal <mdhillman@cmclinnovations.com>}
+ */
+@Controller
+@WebServlet(urlPatterns = {"/get", "/status"})
+public class FeatureInfoAgent extends JPSAgent {
```

Comment:
Seb will disagree on this, but there's no point in using JPSAgent if you're not using processRequestParameters

Reply:
Ok I see now that you are using readJsonParameter, then this is not applicable, documentation somewhere needs to say this takes in a "query" parameter with a JSON string in it